src/index/src/inverted_index/create/sort/intermediate_rw.rs

```diff
@@ -49,10 +49,10 @@ impl<W: AsyncWrite + Unpin> IntermediateWriter<W> {
 
         let value_stream = stream::iter(values.into_iter().map(Ok));
         let frame_write = FramedWrite::new(&mut self.writer, encoder);
+        // `forward()` will flush and close the writer when the stream ends
         value_stream.forward(frame_write).await?;
```

Comment:
> It will complete once the stream is exhausted, the sink has received and flushed all items, and the sink is closed. If the stream produces an error, that error will be returned by this future without flushing/closing the sink.
[ref](https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.forward)

You should handle the err case.

Reply:
Good 👁️ 👁️